### PR TITLE
Add a lidar demo for gmapping and simple drive in the map later

### PR DESCRIPTION
**Pull request by mjstn**
This is meant to be a place to point people to who have questions on running a Lidar on a Magni and are not yet familiar with ROS.   Experts in ROS do not really need this but beginners in NAV will benefit from an exact example to build upon that is built upon the Magni environment and thus avoids all sorts of details specific to other home brew systems.

I avoided having yaml parameter files and so on so that it would be as easy as possible for people to understand what was going on.   Separating out the RPLIDAR launch would be the next thing I would do to avoid it being duplicated in both launch file.  So again, this is a starting point that does work without Sonars/costmap/move_base/amcl.

The launch files are in a simple form that suits Magni environment.  More complex launches to add sonars and have rviz config to show sonars and use move_base and so on can be added to this 'base' that keeps things simple.

### CODE REVIEW

**Comment by mjstn** [comment]
Since this is a demo and not mainstream code and Rohan has been overloaded I am going to ask Janez to take a look at this so that it can make its way into our customers view faster.

File: magni_lidar/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+Our goal here is to put in one place the key elements of robot navigation using a very popular device, a LIDAR, to allow the robot to see walls all around and both map and then later navigate within a known mapped area.
+
+The full system here could be studied and investigated by learning more about such things as ROS /tf topic and the lidar itself as well as how ROS understands the orientation of sensors in 3D space.   Plenty to learn but this is a known starting point as an example.
+
+We will use a relatively low cost and very popular Slamtec  RPLidar A1 that is connected to the Magni raspberry pi USB and will by default in most cases show up as serial device /dev/ttyUSB0.  For these files the RPLidar is screwed to the top plate using 5mm spacers so the ribbon cable can bend around and allow their little USB board to be connected via USB cable to a port of the Raspberry Pi.
+
+The center of the lidar is centered in Y and is thus half way between each wheel but of course on top of the Magni top plate.  the motor and pully is on the rear side of the lidar.
```
Comment (JanezCim) [inline magni_lidar/README.md]: Capital "T" missing :)

File: magni_lidar/launch/magni_lidar_mapmaker.launch [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  <!-- Start the rplidar node built from https://github.com/sharp-rmf/rplidar_ros          -->
+  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
+    <param name="serial_port"         type="string" value="$(arg lidar_serial_port)"/>
+    <param name="serial_baudrate"     type="int"    value="$(arg lidar_serial_baud)"/>
+    <param name="frame_id"            type="string" value="laser"/>
+    <param name="inverted"            type="bool"   value="false"/>
+    <param name="angle_compensate"    type="bool"   value="true"/>
+  </node>
+
+  <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0 0 0.0 /map /odom 20"/>
```
Comment (JanezCim) [inline magni_lidar/launch/magni_lidar_mapmaker.launch:32]: Not critical, but just pointing out that frequency of static publishers can also be smaller then 20Hz

File: magni_lidar/launch/magni_lidar_maprunner.launch [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  <!-- Publish the map to odom transform  as 0 if we are not using amcl                     -->
+  <node pkg="tf" type="static_transform_publisher" name="stp_map_to_odom" args="0.0 0.0 0.0 0 0 0.0 /map /odom 20"/>
+
+  <!-- Optionally set an identity transform so base_footprint is same as base_frame         -->
+  <node pkg="tf" type="static_transform_publisher" name="stp_base_footprint_to_base_frame" args="0.0 0.0 0.0 0 0 0 /base_footprint /base_frame 20"/>
+
+  <!-- publish where base_frame is relative to base_link                                    --> 
+  <node pkg="tf" type="static_transform_publisher" name="stp_base_link_to_base_frame" args="0.0 0.0 -0.1 0 0 0 /base_link /base_frame 20"/>
+
+  <!-- Add in child frame of rplider as it is not in our model yet                          -->
```
Comment (JanezCim) [inline magni_lidar/launch/magni_lidar_maprunner.launch]: rplidar*

File: magni_lidar/package.xml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  <!-- Use build_depend for packages you need at compile time: -->
+  <!--   <build_depend>message_generation</build_depend> -->
+  <!-- Use buildtool_depend for build tool packages: -->
+  <!--   <buildtool_depend>catkin</buildtool_depend> -->
+  <!-- Use run_depend for packages you need at runtime: -->
+  <!--   <run_depend>message_runtime</run_depend> -->
+  <!-- Use test_depend for packages you need only for testing: -->
+  <!--   <test_depend>gtest</test_depend> -->
+  <buildtool_depend>catkin</buildtool_depend>
+
```
Comment (JanezCim) [inline magni_lidar/package.xml:43]: I know in README and launch files you instruct to install it, but would it be smart to have the package dependant on "rplidar_ros" here as well?

**Review by JanezCim — APPROVED** [review]
Compiled without problems, but could not run since im not close to the robot with a rplidar. Since it works for you, i approve :)

File: magni_lidar/README.md (hunk 1 above)
Comment (mjstn) [reply magni_lidar/README.md]: Ok.  There would have been some value in telling me where.  It was the very last sentence so anyway, thanks for the catch.

File: magni_lidar/launch/magni_lidar_mapmaker.launch (hunk 2 above)
Comment (mjstn) [reply magni_lidar/launch/magni_lidar_mapmaker.launch:32]: I had tried earlier things like 5 and 10 but at that time I was having other issues so never changed back.  This is a performance tweek and a valid input, thanks.
I will let it wait as the test of all this is very involved and I never do a blind change of some rate until I can really test using it.  So thanks, next time is the plan for this rate change.

File: magni_lidar/launch/magni_lidar_maprunner.launch (hunk 3 above)
Comment (mjstn) [reply magni_lidar/launch/magni_lidar_maprunner.launch]: Wow, nothing gets by Janez I see.    LOL.   Thanks, will do

File: magni_lidar/package.xml (hunk 4 above)
Comment (mjstn) [reply magni_lidar/package.xml:43]: This too is a good plan.  It will be so.

**Pull request merged**